src/strands/multiagent/swarm.py

```diff
@@ -89,7 +92,17 @@ def __repr__(self) -> str:
         return f"SwarmNode(node_id='{self.node_id}')"
 
     def reset_executor_state(self) -> None:
-        """Reset SwarmNode executor state to initial state when swarm was created."""
+        """Reset SwarmNode executor state to initial state when swarm was created.
+
+        If Swarm is resuming from an interrupt, we reset the executor state from the interrupt context.
+        """
+        if self.swarm and self.swarm._interrupt_state.activated:
+            context = self.swarm._interrupt_state.context[self.node_id]
+            self.executor.messages = context["messages"]
```

Comment:
These 3 sets of state (including setting the private `_interrupt_state`) seem like a sort of checkpointing IMO #1138.  As-is it seems unclean that we're having to manually - and IMHO, it's very weird that we have to set a private member.

I'd imagine it'd be better to do something like:

```python
self.executor.restore_checkpoint(checkpoint=context, restore: { "messages": True, "interrupts": True, "state": True })
```

Thoughts?

(not blocking the PR, but I want to set your gears turning)